lib/src/Utils/File.c

```diff
@@ -332,7 +335,10 @@ cl_program cl_util_read_binaries(const cl_context context,
         for (cl_uint i = 0; i < num_devices; ++i) free(binaries_ptr[i]);
         free(binaries_ptr);
     }
-    free(binaries_size);
+    if(binaries_size != NULL)
+    {
+        free(binaries_size);
+    }
```

Comment:
What prompted this change?  It should be fine to free a NULL pointer.

Reply:
I do not recall what triggered it, but it is true that it should be okay to not check for NULL pointer. I undid this in the [latest commit](https://github.com/KhronosGroup/OpenCL-SDK/pull/88/commits/b3e7aff7e00056508c1051750acb5bc773292f68).